test/integration/kueuectl/list_test.go

```diff
@@ -134,44 +132,48 @@ very-long-local-queue-name   cq1                            0
-			gomega.Expect(errOutput.String()).Should(gomega.BeEmpty())
-			gomega.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME   COHORT   PENDING WORKLOADS   ADMITTED WORKLOADS   ACTIVE   AGE
+				kueuectl.SetArgs([]string{"list", "clusterqueue", "--field-selector",
+					fmt.Sprintf("metadata.name=%s", cq1.Name)})
+				err := kueuectl.Execute()
+				g.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
+				g.Expect(errOutput.String()).Should(gomega.BeEmpty())
+				g.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME   COHORT   PENDING WORKLOADS   ADMITTED WORKLOADS   ACTIVE   AGE
 cq1             0                   0                    true     %s
 `,
-				duration.HumanDuration(executeTime.Sub(cq1.CreationTimestamp.Time)),
-			)))
+					duration.HumanDuration(executeTime.Sub(cq1.CreationTimestamp.Time)),
+				)))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
 		})
 
 		// Simple client set that are using on unit tests not allow paging.
-		ginkgo.It("Should print local queues list with paging", func() {
-			streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
-			configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
-			executeTime := time.Now()
-			kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams, Clock: testingclock.NewFakeClock(executeTime)})
+		ginkgo.It("Should print cluster queues list with paging", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
+				configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
+				executeTime := time.Now()
+				kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams, Clock: testingclock.NewFakeClock(executeTime)})
 
-			os.Setenv(list.KueuectlListRequestLimitEnvName, "1")
-			kueuectl.SetArgs([]string{"list", "clusterqueue"})
-			err := kueuectl.Execute()
+				os.Setenv(list.KueuectlListRequestLimitEnvName, "1")
+				kueuectl.SetArgs([]string{"list", "clusterqueue"})
+				err := kueuectl.Execute()
 
-			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
-			gomega.Expect(errOutput.String()).Should(gomega.BeEmpty())
-			gomega.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME                           COHORT   PENDING WORKLOADS   ADMITTED WORKLOADS   ACTIVE   AGE
+				g.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
+				g.Expect(errOutput.String()).Should(gomega.BeEmpty())
+				g.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME                           COHORT   PENDING WORKLOADS   ADMITTED WORKLOADS   ACTIVE   AGE
 cq1                                     0                   0                    true     %s
-very-long-cluster-queue-name            0                   0                    false    %s
+very-long-cluster-queue-name            0                   0                    true     %s
```

Comment:
Can we reduce the scope of eventually just to the table?

Reply:
Sorry. Which scope do you mean?

Reply:
The eventually essentially does "retry the entire test until it passes". It seems quite aggressive, so I was thinking we could wrap less.

Let me take a step back - what is the reason the test flakes? Please update the PR description. When we understand what is the problem maybe we see a better solution.

Reply:
Active status calculated by `meta.IsStatusConditionTrue(cq.Status.Conditions, v1beta1.ClusterQueueActive)`. Sometime we don't have time to set this condition to `True` between creation clusterqueues (an `JustBeforeEach`) and print list of clusterqueues. So that's why we have active status = false. So we need to wait time to be sure that status set to true.

Reply:
Ah I see, in that case can we await in the `JustBeforeEach` for the CQ to be active? Or just at the beginning of affected tests (`Its`).

Reply:
Good point. Thanks. Fixed